src/app/core/upload/upload.service.ts

```diff
@@ -14,35 +15,40 @@ export class UploadService {
 
   public uploads: Upload[] = [];
 
-  private evaporate: Evaporate;
+  private evaporate: Observable<Evaporate>;
   private bucketName: string = Env.BUCKET_NAME;
   private signUrl: string = Env.SIGN_URL;
   private awsKey: string = Env.AWS_KEY;
   private awsUrl: string = Env.AWS_URL;
   private useCloudfront: boolean = Env.USE_CLOUDFRONT;
 
   constructor(private mimeTypeService: MimeTypeService) {
-    this.init();
+    this.evaporate = this.init();
   }
 
-  init() {
+  init(): Observable<Evaporate> {
     // until there is a good way to load from env and inject
-    this.evaporate = new Evaporate({
-      signerUrl: this.signUrl,
-      aws_key: this.awsKey,
-      aws_url: this.awsUrl,
-      bucket: this.bucketName,
-      cloudfront: this.useCloudfront,
-      onlyRetryForSameFileName: true,
-      logging: false
-    });
+    return Observable.from(
+      Evaporate.create({
+        signerUrl: this.signUrl,
+        aws_key: this.awsKey,
+        aws_url: this.awsUrl,
+        bucket: this.bucketName,
+        cloudfront: this.useCloudfront,
+        onlyRetryForSameFileName: true,
+        logging: false,
+        awsSignatureVersion: '2'
+      }).then(evaporate => evaporate)
+    );
   }
 
-  add(file: File, contentType?: string): Upload {
-    let ct = contentType || this.mimeTypeService.lookupFileMimetype(file).full();
-    let upload = new Upload(file, ct, this.evaporate);
-    this.uploads.push(upload);
-    return upload;
+  add(file: File, contentType?: string): Observable<Upload> {
+    return this.evaporate.map(evaporate => {
```

Comment:
I thought because we had `map`, `from`, and `of` included [from the styleguide](https://github.com/PRX/styleguide.prx.org/blob/master/build.js#L117) that we didn't have to import them in downstream apps.

Reply:
Yeah, it's going to work just fine, since those imports are really global.  I guess the concern is if styleguide drops the operator someday ... and if it's not imported here, it'll break.  Edge case.

I'm not clear what our standard is these days ... I think when they first told us to stop importing `rxjs/*` ... I went through and put imports at the top of every file for operators they used.  I guess my preference is to keep doing that?  But if we've already deviated, i don't feel as strongly.

Reply:
🤷‍♀️ guess I thought we were just importing the ones that aren't global, but I can add them here if we were trying to import wherever used